rust-runtime/aws-smithy-runtime/src/config_bag.rs

```diff
@@ -40,13 +61,40 @@ struct Layer {
     props: PropertyBag,
 }
 
+fn no_op(_: &mut ConfigBag) {}
+
 impl FrozenConfigBag {
+    /// Attempts to convert this bag directly into a [`ConfigBag`] if no other references exist
+    ///
+    /// This allows modifying the top layer of the bag. [`Self::add_layer`] may be
+    /// used to add a new layer to the bag.
     pub fn try_modify(self) -> Option<ConfigBag> {
         Arc::try_unwrap(self.0).ok()
     }
 
-    #[must_use]
-    pub fn with_open(&self, name: &'static str, next: impl Fn(&mut ConfigBag)) -> ConfigBag {
+    /// Add a new layer to the config bag
+    ///
+    /// This is equivalent to calling [`Self::with_fn`] with a no-op function
+    ///
+    /// # Examples
+    /// ```
+    /// fn add_more_config(bag: &mut ConfigBag) { /* ... */ }
+    /// use aws_smithy_runtime::ConfigBag;
+    /// let bag = ConfigBag::base().with_fn("first layer", |_| { /* add a property */ });
+    /// let mut bag = bag.add_layer("second layer");
+    /// add_more_config(&mut bag);
+    /// let bag = bag.freeze();
+    /// ```
+    pub fn add_layer(&self, name: &'static str) -> ConfigBag {
+        self.with_fn(name, no_op)
+    }
```

Comment:
`/// let mut bag = bag.add_layer("second layer");` doesn't compile because `ConfigBag` doesn't have an `add_layer` method. Did you mean to define this on `ConfigBag` instead of `FrozenConfigBag`?

Reply:
ah, should have a `bag.freeze().add_layer`—we can also add `add_layer` to `ConfigBag` as well but moving self